src/components/dataTable/js/tableRowDirective.js

```diff
@@ -0,0 +1,326 @@
+  }
+
+  this.$$toggleAutoSelectFn = function() {
+    this.$scope.$evalAsync(this.toggleSelection.bind(this));
+  }.bind(this);
+
+  this.selectIdentifier = expressionSplit[1];
+  this.selectIdentifierAlias = expressionSplit[2];
+
+  if (this.$attrs.mdTrackBy) {
+    var identifier = this.selectIdentifierAlias || this.selectIdentifier;
+    this.compareTrackFn = this.createTrackByFn(identifier, this.$attrs.mdTrackBy);
+  }
+
+  // Watch in the parent scope, because the mdTable directive, creates its own scope.
+  this.$scope.$parent.$watch(this.selectIdentifier, function(newItem) {
+    this.itemObject = newItem;
+
+    if (this.selectIdentifierAlias) {
+      // When the user specifies an alias, we will add it to their scope, so they can use the alias property
+      // in the template.
+      this.$scope.$parent[this.selectIdentifierAlias] = newItem;
+    }
+  }.bind(this));
+
+  this.$attrs.$observe('disabled', function(isDisabled) {
+    this.isDisabled = $mdUtil.parseAttributeBoolean(isDisabled);
+  }.bind(this));
+}
+
+MdTableSelectController.prototype.initCtrl = function(tableCtrl, rowCtrl) {
+  this.table = tableCtrl;
+  this.rowCtrl = rowCtrl;
+};
+
+MdTableSelectController.prototype.createTrackByFn = function(identifier, trackExpression) {
+  var locals = {};
+  var trackByFn = this.$parse(trackExpression);
+
+  return function(item) {
+    locals[identifier] = item;
+
+    return trackByFn(this.$scope, locals);
+  }.bind(this);
+};
+
+MdTableSelectController.prototype._compileCheckbox = function() {
+  return this.$compile(
+    '<md-checkbox aria-label="Select Row" ng-click="$mdSelectable.onCheckboxToggle($event)" ' +
+                 'ng-checked="$mdSelectable.isSelected()" ng-disabled="$mdSelectable.isDisabled">'
+  )(this.$scope);
+};
+
+MdTableSelectController.prototype.isEqualItem = function(compareObj) {
+  return this.compareTrackFn ?
+         this.compareTrackFn(compareObj) === this.compareTrackFn(this.itemObject) :
+         this.itemObject === compareObj;
+};
+
+MdTableSelectController.prototype.setSelectable = function(state) {
+  if (state) {
+    angular.element(this.rowCtrl.getCells(0)).append(this._compileCheckbox());
+
+    this.toggleAutoSelect(this.$attrs.mdAutoSelect);
+
+    this.selectedWatcher = this.$scope.$watch(this.isSelected.bind(this), this.onSelectionChange.bind(this));
+    this.autoSelectWatcher = this.$attrs.$observe('mdAutoSelect', this.toggleAutoSelect.bind(this));
+  } else {
+    this.selectedWatcher();
+    this.autoSelectWatcher();
+
+    this.$element.removeClass('md-selected');
+
+    if (this.$element.hasClass('md-auto-select')) {
+      this.toggleAutoSelect(false);
+    }
+  }
+};
+
+MdTableSelectController.prototype.toggleAutoSelect = function(state) {
+  state = this.$mdUtil.parseAttributeBoolean(state);
+
+  this.$element.toggleClass('md-auto-select', !!state);
+
+  if (state) {
+    this.$element.on('click', this.$$toggleAutoSelectFn);
```

Comment:
_Same as said above_ 

Click should be properly triggered on touch devices as well. 

> Especially with the Fastclick solutions we use ($mdGesture and soon HammerJS)